### PR TITLE
Added TagManager.dom.onClick functionality and clickButton as pre configured variable

### PR DESCRIPTION
**Pull request by AltamashShaikh**
Added TagManager.dom.onClick functionality and clickButton as pre configured variable

### Description:

Added TagManager.dom.onClick functionality and clickButton as pre configured variable
Issue: #268 

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by tsteur** [comment]
@AltamashShaikh great. Tested it and worked nicely.  The only change I would suggest is to mention it in the description of the trigger that this will trigger on left, middle and right click as it might not be expected. Could also mention that this may be restricted using the button variable.
image

Let's also add a new FAQ for this like "How do I get the click triggers to only trigger on left, middle or right click?". It is otherwise not obvious to which variable needs to be used and which values to restrict to for example only a left click.

I'll send you instructions internally. The FAQ should appear [here](https://matomo.org/faq/tag-manager/) by assigning the parent Tag Manager category and entering an "order/sort" like 15.

File: lang/en.json [hunk 1]
```diff
@@ -170,8 +172,8 @@
         "DomReadyTriggerName": "DOM Ready",
         "DriftDescription": "Drift allows you to connect with your customers.",
         "DriftHelp": "This tag allows you to add the Drift contact form to your website.",
-        "AllDownloadsClickTriggerDescription": "Triggered when a link is clicked which links to a downloadable file.",
-        "AllDownloadsClickTriggerHelp": "Triggered when a user clicks on an \"A\" or \"AREA\" element and the link points to a file with a downloadable file extension.",
+        "AllDownloadsClickTriggerDescription": "Triggered when a link is clicked which links to a downloadable file. It will be triggered on left,middle and right click.",
```
Comment (tsteur) [inline lang/en.json]: there's a space missing after the comma @AltamashShaikh

File: lang/en.json [hunk 2]
```diff
@@ -170,8 +172,8 @@
...
+        "AllDownloadsClickTriggerHelp": "Triggered when a user clicks on an \"A\" or \"AREA\" element and the link points to a file with a downloadable file extension. To enable tracking on specific click button, please add conditions based on a \"click\" variable in the advanced settings.",
```
Comment (tsteur) [inline lang/en.json]: maybe you mean "ClickButton" variable?

File: lang/en.json (hunk 1 above)
Comment (tsteur) [reply lang/en.json]: same also for the others below

File: lang/en.json (hunk 2 above)
Comment (tsteur) [reply lang/en.json]: same also for the others below

**Review by tsteur — CHANGES_REQUESTED** [review]

File: lang/en.json (hunk 2 above)
Comment (AltamashShaikh) [reply lang/en.json]: Updated

File: lang/en.json (hunk 1 above)
Comment (AltamashShaikh) [reply lang/en.json]: Updated

**Review by tsteur — APPROVED** [review]
@AltamashShaikh looks good to merge as soon as the FAQ has been published.

**Pull request merged**